wal/decoder.go

```diff
@@ -98,6 +113,44 @@ func (d *decoder) decodeRecord(rec *walpb.Record) error {
 	return nil
 }
 
+// isTornEntry determines whether the last entry of the WAL was partially written
+// and corrupted because of a torn write.
+func (d *decoder) isTornEntry(data []byte) bool {
```

Comment:
do we need to do this actually? If we can ensure the size is always written correctly, then we can probably just skip this record and looks for the next one. If the next one does not exist, then it is a partial write. If it is a partial write, we can clean size bytes and len(size) bytes after it.